api/v1alpha1/nodepool_types.go

```diff
@@ -316,6 +316,52 @@ type NodePoolPlatform struct {
 	Agent *AgentNodePoolPlatform `json:"agent,omitempty"`
 
 	Azure *AzureNodePoolPlatform `json:"azure,omitempty"`
+
+	// PowerVS specifies the configuration used when using IBMCloud PowerVS platform.
+	//
+	// +optional
+	PowerVS *PowerVSNodePoolPlatform `json:"powervs,omitempty"`
+}
+
+// PowerVSNodePoolPlatform specifies the configuration of a NodePool when operating
+// on IBMCloud PowerVS platform.
+type PowerVSNodePoolPlatform struct {
+	// ServiceInstanceID is the ServiceInstance to use for control plane cloud resources.
+	ServiceInstanceID string `json:"serviceInstanceID"`
+
+	// SysType used to host the instance(e.g: s922, e980, e880)
+	// +optional
+	SysType string `json:"sysType,omitempty"`
+
+	// ProcType (dedicated, shared, capped)
+	// +optional
+	ProcType string `json:"procType,omitempty"`
```

Comment:
please add open api enum validation tag

Reply:
we have a web hook in the capibm controller to set some of these value default, rightnow I'm not deploying that part of the deployment(future plan is to add support).

So my next question is how are we dealing with other providers?

Reply:
I don't think that should make any difference, I see webhooks as an additional UX layer. The API need to be solid even nevertheless. If a ProcType is necessary for the controller to work, then the field must be  mark as required not optional. Then in the CLI we can give it a happy path/convenience value by default if we want to.

Reply:
sure, I will make enum and default.